parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -932,12 +932,48 @@ public ParquetFileReader(InputFile file, ParquetReadOptions options, SeekableInp
     this.options = options;
     try {
       this.footer = readFooter(file, options, f, converter);
+    } catch (IOException e) {
+      f.close();
+      throw e;
+    }
+
+    this.fileMetaData = footer.getFileMetaData();
+    this.fileDecryptor = fileMetaData.getFileDecryptor(); // must be called before filterRowGroups!
+    if (null != fileDecryptor && fileDecryptor.plaintextFile()) {
+      this.fileDecryptor = null; // Plaintext file. No need in decryptor
+    }
+
+    try {
+      this.blocks = filterRowGroups(footer.getBlocks());
     } catch (Exception e) {
-      // In case that reading footer throws an exception in the constructor, the new stream
+      // In case that filterRowGroups throws an exception in the constructor, the new stream
       // should be closed. Otherwise, there's no way to close this outside.
       f.close();
       throw e;
     }
+    this.blockIndexStores = listWithNulls(this.blocks.size());
+    this.blockRowRanges = listWithNulls(this.blocks.size());
+    for (ColumnDescriptor col : footer.getFileMetaData().getSchema().getColumns()) {
+      paths.put(ColumnPath.get(col.getPath()), col);
+    }
+
+    if (options.usePageChecksumVerification()) {
+      this.crc = new CRC32();
+      this.crcAllocator = ReusingByteBufferAllocator.strict(options.getAllocator());
+    } else {
+      this.crc = null;
+      this.crcAllocator = null;
+    }
+  }
+
+  public ParquetFileReader(InputFile file, ParquetMetadata footer, ParquetReadOptions options, SeekableInputStream f)
```

Comment:
Could you add a javadoc for this new method?

Reply:
Please add the corresponding `static open` method together